winit/src/application.rs

```diff
@@ -179,41 +183,81 @@ where
-                        new_inner_size,
-                        ..
-                    },
-                window_id,
-            } => Some(winit::event::Event::WindowEvent {
-                event: winit::event::WindowEvent::Resized(*new_inner_size),
-                window_id,
-            }),
-            _ => event.to_static(),
-        };
-
-        if let Some(event) = event {
-            sender.start_send(event).expect("Send event");
-
-            let poll = instance.as_mut().poll(&mut context);
-
-            *control_flow = match poll {
-                task::Poll::Pending => ControlFlow::Wait,
-                task::Poll::Ready(_) => ControlFlow::Exit,
+            let event = match event {
+                winit::event::Event::WindowEvent {
+                    event:
+                        winit::event::WindowEvent::ScaleFactorChanged {
+                            new_inner_size,
+                            ..
+                        },
+                    window_id,
+                } => Some(winit::event::Event::WindowEvent {
+                    event: winit::event::WindowEvent::Resized(*new_inner_size),
+                    window_id,
+                }),
+                _ => event.to_static(),
             };
-        }
-    });
 
-    Ok(())
+            if let Some(event) = event {
+                sender.start_send(event).expect("Send event");
+
+                let poll = instance.as_mut().poll(&mut context);
+
+                *control_flow = match poll {
+                    task::Poll::Pending => ControlFlow::Wait,
+                    task::Poll::Ready(_) => ControlFlow::Exit,
+                };
+            }
+        });
+
+        Ok(())
+    }
+
+    #[cfg(target_arch = "wasm32")]
+    {
+        event_loop.run(move |event, _, control_flow| {
+            use winit::event_loop::ControlFlow;
+
+            if let ControlFlow::Exit = control_flow {
+                return;
+            }
+
+            let event = match event {
+                winit::event::Event::WindowEvent {
+                    event:
+                        winit::event::WindowEvent::ScaleFactorChanged {
+                            new_inner_size,
+                            ..
+                        },
+                    window_id,
+                } => Some(winit::event::Event::WindowEvent {
+                    event: winit::event::WindowEvent::Resized(*new_inner_size),
+                    window_id,
+                }),
+                _ => event.to_static(),
+            };
+
+            if let Some(event) = event {
+                sender.start_send(event).expect("Send event");
+
+                let poll = instance.as_mut().poll(&mut context);
+
+                *control_flow = match poll {
+                    task::Poll::Pending => ControlFlow::Wait,
+                    task::Poll::Ready(_) => ControlFlow::Exit,
+                };
+            }
+        });
+    }
```

Comment:
I'm trying to understand why these two different code paths are needed. Can you explain?

I will add a comment or, ideally, try to merge them somewhat.

Reply:
Actually, yes, this can be de-duplicated, I just had no enough time to do it, so I did it quick and dirty \*sorry\* and then forgot to fix this code chunk. @kaimast If you interested in this PR, I sent your invite to be a collaborator on the fork, you welcome to make direct changes to the PR.